sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,154 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import reactor.core.publisher.Flux;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
+import java.net.ProtocolException;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * A socket client used for making PATCH requests
+ */
+
+class SocketClient {
+
+    private static final String HTTP_VERSION = " HTTP/1.1";
+
+    /**
+     * Opens a socket connection, then writes the PATCH request across the
+     * connection and reads the response
+     *
+     * @param httpRequest {@link com.azure.core.http.HttpRequest} instance
+     * @return an instance of HttpUrlConnectionResponse
+     */
+    public static HttpUrlConnectionResponse sendPatchRequest(HttpRequest httpRequest) throws IOException {
+        String host = httpRequest.getUrl().getHost();
+        int port = httpRequest.getUrl().getPort();
+        if (httpRequest.getUrl().getProtocol().equals("https")) {
+            SSLSocketFactory sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            try (SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        } else if (httpRequest.getUrl().getProtocol().equals("http")) {
+            try (Socket socket = new Socket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        }
```

Comment:
Try to reduce repeating yourself. Also, do you need to call `SSLSocketFactory.getDefault()` every time, or can you cache it?

```suggestion
        final URL requestUrl = httpRequest.getUrl();
        final String protocol = requestUrl.getProtocol();
        final String host = requestUrl.getHost();
        final int port = requestUrl.getPort();
        
        switch (protocol) {
            case "https": {
                SSLSocketFactory sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
                try (SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(host, port)) {
                    return doInputOutput(httpRequest, socket);
                }
            }
            case "http": {
                try (Socket socket = new Socket(host, port)) {
                    return doInputOutput(httpRequest, socket);
                }
            }
        }
```

Reply:
This was marked resolved but looks unchanged - reopening.